dbms/src/Storages/Transaction/TiDB.cpp

```diff
@@ -110,14 +110,22 @@ Field ColumnInfo::defaultValueToField() const
     }
     switch (tp)
     {
-    // TODO: Consider unsigned?
     // Integer Type.
     case TypeTiny:
     case TypeShort:
     case TypeLong:
     case TypeLongLong:
     case TypeInt24:
-        return value.convert<Int64>();
+    {
+        try
+        {
+            return value.convert<Int64>();
+        }
+        catch (...)
+        {
+            return static_cast<Int64>(std::stoull(value.convert<String>()));
```

Comment:
Should we fix at TiDB side instead? IMO it is wired for a int column to pass a "string" default value. The casting rule may be very complicated as well.

Reply:
I agree with you. https://github.com/pingcap/tiflash/issues/3157#issuecomment-1128606869. But we maybe still need to add a backstop strategy to support upgrade https://github.com/pingcap/tiflash/issues/3157#issuecomment-1128615478

Reply:
If the value is "1.9", it should be parsed into integer "2". Verified under TiDB and MySQL.

```
mysql> select version();
+-------------------------+
| version()               |
+-------------------------+
| 5.7.33-0ubuntu0.16.04.1 |
+-------------------------+
1 row in set (0.00 sec)

mysql> create table td ( a int default 1.9, b int);
Query OK, 0 rows affected (0.20 sec)

mysql> show create table td;
+-------+------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                     |
+-------+------------------------------------------------------------------------------------------------------------------+
| td    | CREATE TABLE `td` (
  `a` int(11) DEFAULT '2',
  `b` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=latin1 |
+-------+------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> insert into td(b) values(1);
Query OK, 1 row affected (0.01 sec)

mysql> select * from td;
+------+------+
| a    | b    |
+------+------+
|    2 |    1 |
+------+------+
1 row in set (0.00 sec)
```

```
test> select tidb_version()
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                             |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v6.0.0\nEdition: Community\nGit Commit Hash: 36a9810441ca0e496cbd22064af274b3be771081\nGit Branch: heads/refs/tags/v6.0.0\nUTC Build Time: 2022-03-31 10:33:28\nGoVersion: go1.18\nRace Enabled: false\nTiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306\nCheck Table Before Drop: false |
+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.018s
test> drop table td;
You're about to run a destructive command.
Do you want to proceed? (y/n): y
Your call!
Query OK, 0 rows affected
Time: 0.197s
test> create table td ( a int default 1.9, b int);
Query OK, 0 rows affected
Time: 0.079s
test> show create table td;
+-------+--------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                               |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------+
| td    | CREATE TABLE `td` (\n  `a` int(11) DEFAULT '1.9',\n  `b` int(11) DEFAULT NULL\n) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+--------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set
Time: 0.018s
test> insert into td(b) values(1);
Query OK, 1 row affected
Time: 0.009s
test> select * from td;
+---+---+
| a | b |
+---+---+
| 2 | 1 |
+---+---+
1 row in set
Time: 0.022s
```